### PR TITLE
unexport ensureHostPort and DRY ports

### PR DESCRIPTION
**Pull request by tamird**
We no longer accept naked address arguments on the command line except
for use in gossip resolvers, so ensureHostPort can be limited to the
gossip/resolver package.

Also reduces the duplication of "25267" and "15432" in the codebase.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4161)

### CODE REVIEW

File: base/context.go [hunk 1]
```diff
@@ -37,6 +37,12 @@ const (
 	httpScheme      = "http"
 	httpsScheme     = "https"
 
+	// Default HTTP[S]/RPC[S] port.
+	CockroachPort = "26257"
+
+	// Default PG port.
+	PgPort = "15432"
```
Comment (tschottdorf) [inline base/context.go]: should this be PGPort?

File: gossip/resolver/resolver.go [hunk 2]
```diff
@@ -99,3 +100,29 @@ func NewResolverFromAddress(addr net.Addr) (Resolver, error) {
 func NewResolverFromUnresolvedAddr(addr util.UnresolvedAddr) (Resolver, error) {
 	return NewResolverFromAddress(&addr)
 }
+
+// ensureHostPort takes a host:port pair, where the host and port are optional.
+// If host and port are present, the output is equal to the input. If port is
+// not present, use default port 26257(cockroach) or 15432(postgres). If host is
```
Comment (tschottdorf) [inline gossip/resolver/resolver.go]: defaultPort is passed in, so why spell them out here?

**Comment by tschottdorf** [comment]
LGTM. The `host` parameter's behavior hasn't been affected, right?

**Comment by tamird** [comment]
Correct.

---

Review status: 0 of 14 files reviewed at latest revision, 2 unresolved discussions.

---

*[base/context.go, line 44 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4161#-K9lqmQ4hBDcZFHRzfUx-r1-44:-K9m8kVenXzipkeQJWKf:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/26ab0208487cb80a047f6f4bcb19e09a4292f6bf/base/context.go#L44)):*
Done.

---

*[gossip/resolver/resolver.go, line 106 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4161#-K9lqmQ4hBDcZFHRzfV1-r1-106:-K9m8lVC_JFQ2vjL7SqL:1844639070) ([raw file](https://github.com/cockroachdb/cockroach/blob/26ab0208487cb80a047f6f4bcb19e09a4292f6bf/gossip/resolver/resolver.go#L106)):*
Rewrote this comment.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4161#-:-K9m8iklXnM3hGIYT1v1:1798441257)*

**Comment by RaduBerinde** [comment]
LGTM

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4161#-:-K9mgHUtng3TlWr5DhPb:676834233)*

**Pull request merged**